agent/proxycfg/upstreams.go

```diff
@@ -92,55 +90,60 @@ func (s *handlerUpstreams) handleUpdateUpstreams(ctx context.Context, u cache.Up
 		}
 		upstreamsSnapshot.WatchedUpstreamEndpoints[uid][targetID] = resp.Nodes
 
-		var passthroughAddrs map[string]ServicePassthroughAddrs
+		if s.kind != structs.ServiceKindConnectProxy || s.proxyCfg.Mode != structs.ProxyModeTransparent {
+			return nil
+		}
 
-		for _, node := range resp.Nodes {
-			if snap.Proxy.Mode == structs.ProxyModeTransparent && node.Service.Proxy.TransparentProxy.DialedDirectly {
-				if passthroughAddrs == nil {
-					passthroughAddrs = make(map[string]ServicePassthroughAddrs)
+		if _, ok := upstreamsSnapshot.PassthroughUpstreams[uid]; ok {
```

Comment:
I think this block needs a comment, because I am reading this and it seems possibly wrong, or I haven't grokked what is supposed to happen here:
> for all node addresses associated with this  target, if there's an index entry for the address that matches our target and upstream (so it's our own) then _delete it (?)_

Reply:
Is the intent to first "clear out" this upstreamid+targetid from `PassthroughIndices` and `PassthroughUpstreams` and then repopulate them?

Reply:
Yeah it was to clean up data from service instances that aren't in the latest response. I added a comment to that effect.